src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -13,6 +13,40 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     internal sealed class SourcePropertySymbol : SourcePropertySymbolBase
     {
+        internal bool ContainsFieldKeyword
+        {
+            get
+            {
+                if (CSharpSyntaxNode is not PropertyDeclarationSyntax property)
+                {
+                    return false;
+                }
+
+                if (property.AccessorList is not { } accessorList)
+                {
+                    return false;
+                }
+
+                foreach (var accessor in accessorList.Accessors)
+                {
+                    var body = ((SyntaxNode?)accessor.Body ?? accessor.ExpressionBody);
+                    if (body is null)
+                    {
+                        continue;
+                    }
+
+                    var containsFieldKeyword = body.DescendantTokens()
+                        .Any(t => t.IsKind(SyntaxKind.IdentifierToken) && t.ContextualKind() == SyntaxKind.FieldKeyword && !t.Parent.IsKind(SyntaxKind.AttributeTargetSpecifier));
+                    if (containsFieldKeyword)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
```

Comment:
Worth caching?

Reply:
> Worth caching?

Probably. As I mentioned earlier, we prefer to avoid touching syntax unless absolutely necessary. This can lead to re-parsing. If we are doing this for every property in the program, that can have a noticeable negative impact. I would investigate an approach that captures this information when we create symbols for the accessors because the syntax is already available then. Also we could capture the information per specific accessor and, perhaps, avoid binding an accessor that doesn't use the keyword.